homeassistant/components/withings/calendar.py

```diff
@@ -0,0 +1,104 @@
+"""Calendar platform for Withings."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from datetime import datetime
+
+from aiowithings import WithingsClient, WorkoutCategory
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.helpers.entity_registry as er
+
+from . import DOMAIN, WithingsData
+from .coordinator import WithingsWorkoutDataUpdateCoordinator
+from .entity import WithingsEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the calendar platform for entity."""
+    ent_reg = er.async_get(hass)
+    withings_data: WithingsData = hass.data[DOMAIN][entry.entry_id]
+
+    workout_coordinator = withings_data.workout_coordinator
+
+    calendar_setup_before = ent_reg.async_get_entity_id(
+        Platform.CALENDAR,
+        DOMAIN,
+        f"withings_{entry.unique_id}_workout",
+    )
+
+    if workout_coordinator.data is not None or calendar_setup_before:
+        async_add_entities(
+            [WithingsWorkoutCalendarEntity(withings_data.client, workout_coordinator)],
+        )
+    else:
+        remove_calendar_listener: Callable[[], None]
+
+        def _async_add_calendar_entity() -> None:
+            """Add calendar entity."""
+            if workout_coordinator.data is not None:
+                async_add_entities(
+                    [
+                        WithingsWorkoutCalendarEntity(
+                            withings_data.client, workout_coordinator
+                        )
+                    ],
+                )
+                remove_calendar_listener()
+
+        remove_calendar_listener = workout_coordinator.async_add_listener(
+            _async_add_calendar_entity
+        )
+
+
+def get_event_name(category: WorkoutCategory) -> str:
+    """Return human-readable category."""
+    name = category.name.lower().capitalize()
+    return name.replace("_", " ")
+
+
+class WithingsWorkoutCalendarEntity(CalendarEntity, WithingsEntity):
+    """A calendar entity."""
+
+    _attr_translation_key = "workout"
+
+    coordinator: WithingsWorkoutDataUpdateCoordinator
+
+    def __init__(
+        self, client: WithingsClient, coordinator: WithingsWorkoutDataUpdateCoordinator
+    ) -> None:
+        """Create the Calendar entity."""
+        super().__init__(coordinator, "workout")
+        self.client = client
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+        return None
```

Comment:
Why aren't you return the next upcoming one?

Reply:
You can only see old workouts. Maybe there is a way to track a current workout, but that's not something I am aware of